### ISSUE TITLE
ubuntu-core-initramfs v66.1 and 66 missing libs on ARM64

### ISSUE THREAD

**Issue by RuhanSA079** [snapcore/core-initrd#166]
Good day,

It appears that building Ubuntu Core 20 image with custom kernel is missing the libpthread.so.0 library, using latest 66.1 arm64 deb package from ubuntu image PPA. (Jammy?)

Booting with snapd/edge version -> 2.59.4+git895.g5aeeeae
```
         Starting Wait for the Ubuntu Core chooser trigger...
[    5.446717] caam 30900000.crypto: device ID = 0x0a16040100000100 (Era 9)
[    5.455275] caam 30900000.crypto: job rings = 1, qi = 0
[    5.408604] systemd[1]: Condition check resulted in Daily Cleanup of Temporary Directories being skipped.
[FAILED] Failed to start Wait for the Ubuntu Core chooser trigger.
See 'systemctl status snapd.recovery-chooser-trigger.service' for details.
[    5.485836] systemd[1]: Reached target Basic System.
[    5.528596] systemd[1]: Reached target Timer Units.
[    5.540693] snap-bootstrap[294]: @snap-bootstrap: error while loading shared libraries: libpthread.so.0: cannot open shared object file: No such file or directory
[    5.565044] systemd[1]: Condition check resulted in Show Plymouth Boot Screen being skipped.
[    5.584506] systemd[1]: Starting Wait for the Ubuntu Core chooser trigger...
[    5.600359] systemd[1]: snapd.recovery-chooser-trigger.service: Main process exited, code=exited, status=127/n/a
[    5.620379] systemd[1]: snapd.recovery-chooser-trigger.service: Failed with result 'exit-code'.
[    5.640325] systemd[1]: Failed to start Wait for the Ubuntu Core chooser trigger.
[    5.656818] systemd-udevd[282]: Using default interface naming scheme 'v249'.
[    5.672802] systemd-udevd[283]: Using default interface naming scheme 'v249'.
```

Checking inside that deb, it appears that libpthread is not present, but present in a older ubuntu-core-initramfs v55 package.

@alfonsosanchezbeato  Could you verify this on a ARM64 platform?

**Comment by xnox** [comment]
@RuhanSA079  are you modifying any portions of ubuntu-core-initramfs? are you recompiling snap-bootstrap? are you compiling that in a matching userspace?

**Comment by xnox** [comment]
go binaries are release specific, ubuntu-core-initramfs is release specific, one cannot inject mix & match go binaries between releases and builds.

**Comment by RuhanSA079** [comment]
> @RuhanSA079 are you modifying any portions of ubuntu-core-initramfs? are you recompiling snap-bootstrap? are you compiling that in a matching userspace?

I'm compiling my kernel in a matching userspace (Focal Multipass VM, amd64) for core20.
I'm not modifying anything related to ubuntu-core-initramfs, only the snapcraft kernel plugin to look at a local ubuntu-core-initrd deb file, for it's initrd creation and whatnot.

I think what I did wrong, is used a Jammy ubuntu-core-initrd deb package, not a focal one.
I tried another deb,it appears to be working now, sorry for the noise.

Lesson learned: Use the correct debs from the specific release for the specific version of Ubuntu Core (Focal for Core20, Jammy for Core22)
Too bad that the FDE stuff is still broken somehow.
Another topic for another day.
Once again, sorry for the noise.
